### PR TITLE
startup: catch errors in build_modules_xml.py, grass.py

### PR DESCRIPTION
**Pull request by neteler**
This PR is an extract from #348 (extracted and updated with `black`):

- gui/wxpython/tools/build_modules_xml.py edits:
    - https://github.com/OSGeo/grass/pull/348/files#diff-11ca77721b6d009b6537c54be4172efed843021797676258ad1141a6c63e6fd1
- lib/init/grass.py edit:
   - https://github.com/OSGeo/grass/pull/348/files#diff-647be4ef599868c33f2f69f0d899f79f0aee8d75d1db74f523268c1ac94cddf7

Not sure if we want these changes but just to avoid to see them lost when closing #348 in favour of #2684.

### CODE REVIEW

File: lib/init/grass.py [hunk 1]
```diff
@@ -769,7 +769,8 @@ def set_paths(grass_config_dir):
 
     # Set LD_LIBRARY_PATH (etc) to find GRASS shared libraries
     # this works for subprocesses but won't affect the current process
-    path_prepend(gpath("lib"), LD_LIBRARY_PATH_VAR)
+    if not LD_LIBRARY_PATH_VAR == "":
```
Comment (wenzeslaus) [inline lib/init/grass.py:772]: Simpler:

```suggestion
    if LD_LIBRARY_PATH_VAR:
```

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
Generally, this looks good.

I'm not sure about managing these small changes smartly for the release notes. It does not fit into these categories:

https://github.com/OSGeo/grass/blob/main/utils/release.yml

**Comment by neteler** [comment]
> I'm not sure about managing these small changes smartly for the release notes.

I have changed to use the "startup: ..." keyword.

**Comment by neteler** [comment]
@wenzeslaus may I merge this PR? Not sure how to properly dismiss the "Changes requested" which I believe have been addressed.

**Review by wenzeslaus — APPROVED** [review]

**Pull request merged**